src/frequenz/sdk/microgrid/_data_pipeline.py

```diff
@@ -119,26 +119,52 @@ def __init__(
         self._battery_pools: dict[frozenset[int], BatteryPoolReferenceStore] = {}
         self._frequency_pool: dict[int, GridFrequency] = {}
 
-    def frequency(self, component: Component | None = None) -> GridFrequency:
+    @typing.overload
+    def frequency(self, component_or_id: None = None) -> GridFrequency:
+        ...
+
+    @typing.overload
+    def frequency(self, component_or_id: int) -> GridFrequency:
+        ...
+
+    @typing.overload
+    def frequency(self, component_or_id: Component) -> GridFrequency:
+        ...
+
+    def frequency(
+        self, component_or_id: Component | int | None = None
```

Comment:
I would also keep the variable name as `component`, as what you want to pass is a component, which could be identified by an object or an ID, to make it clear I think eventually we should have a `ComponentId` type that is not compatible with `int`, as we don't want people also to be able to add IDs, etc. That would make this method signature (and others) much clearer.

Also changing the argument name is technically a breaking change, as people could be using `frequency(component=...)`. We should use the syntax `def frequency(self, /, component_or_id: ...)` to force users not to use the keyword syntax for stuff where it doesn't make a lot of sense.

Finally, if we were going to rename this argument, I would call it `source`, as `component` is describing (or was) the type, and not what the argument represents.

Aaaaaaanyway, all that said, for this PR I would be happy with any combination of these options:
* Leave `component` as it was before
* Rename it to `source` and release notes about the breaking change (new name)
* Adding `/` and release notes about the breaking change (drop support to use keyword syntax)

Reply:
I created a discussion around `ComponentId`.